### ISSUE TITLE
On the issue of uploading test set prediction labels to codalab

### ISSUE THREAD

**Issue by completezealous** [PRBonn/semantic-kitti-api#144]
Hello! 
I uploaded semanticKitti's test set predicted labels 11-21 to codalab, The result zip file has passed the validation of validate_submission.py in SemanticKITTI API.
'''
(semanticvisualize) lk@sun:~/semantic-kitti-api$ ./validate_submission.py --task segmentation /home/lk/sequences_2.zip /data/SemanticKITTI/dataset
Validating zip archive "/home/lk/sequences_2.zip".

 ============ segmentation ============ 
  1. Checking filename.............. ✓
  2. Checking directory structure... ✓
  3. Checking file sizes............ ✓

Everything ready for submission! 
'''
 I successfully used semantickitt-api, Visualizes test sets 11-21 with predicted labels, which means my zip file structure is OK and the predicted labels are valid, but I still fail to upload to codalab. (The download file from codalab is also complete)
visualizes display:
![image](https://github.com/PRBonn/semantic-kitti-api/assets/56358219/dc17a6dc-8dc3-46bf-8842-57b195d3545e)

display error:
![image](https://github.com/PRBonn/semantic-kitti-api/assets/56358219/26cc256b-c0f4-4369-ad1d-78881ecef7aa)
the error is the same all four times.
 I honestly don't know why this is happening, because the validate_submitt.py and visualization above show that the file structure and labels I uploaded are valid.

**Comment by jbehley** [comment]
Sorry for the inconvenience, I will have a look what might be wrong.

What I don't check: the number of files. so if you have additional files in the folders that might trigger the condition.

Note that failed submissions don't count towards the max number of submissions.

**Comment by completezealous** [comment]
> Sorry for the inconvenience, I will have a look what might be wrong.
> 
> What I don't check: the number of files. so if you have additional files in the folders that might trigger the condition.
> 
> Note that failed submissions don't count towards the max number of submissions.

Thank you for your reply, but there are no additional files in the folder

**Comment by completezealous** [comment]
> Sorry for the inconvenience, I will have a look what might be wrong.
> 
> What I don't check: the number of files. so if you have additional files in the folders that might trigger the condition.
> 
> Note that failed submissions don't count towards the max number of submissions.

Now,I also used evaluate_semantics.py in your semantic-kitti-api to verify the predicted labels of the test set 11-21. Of course, I did not have the real labels of the test set, and I used the predicted labels as the real labels. Although the final iou is not correct, the side also shows that all my predicted labels are valid. However, when my predicted labels are uploaded to codalab for prediction, I still get the same error as before. Is it because your code misjudged that my labels are training set labels, which leads to an inconsistent number of them? Or is it something else?
![image](https://github.com/PRBonn/semantic-kitti-api/assets/56358219/897fca5c-ed72-4964-a208-708431dd7f5c)
 In addition, I hope you add code to evaluate_semantics.py in codalab to print the number of labels and predict the number of labels.

**Comment by jbehley** [comment]
It seems like you are missing some files in your local KITTI copy.

![image](https://github.com/PRBonn/semantic-kitti-api/assets/11506664/5535b57c-4436-4003-b230-6fd92bae1970)

We are currently just checking if the label files are consistent with your bin files.

But I will modify the script to make this more explicit also on the codalab evaluation script.

**Comment by jbehley** [comment]
I added no some more diagnostics to the evaluation script that puts out some output like this (this is for one of your submissions): 

![image](https://github.com/PRBonn/semantic-kitti-api/assets/11506664/92b61e11-e99e-4b9e-9bba-587e884be238)

Thus, there seem to be some missing files from extracting the original KITTI data. (Or some label files don't get written.)

**Comment by completezealous** [comment]
> I added no some more diagnostics to the evaluation script that puts out some output like this (this is for one of your submissions):
> 
> ![image](https://private-user-images.githubusercontent.com/11506664/309251973-92b61e11-e99e-4b9e-9bba-587e884be238.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkyOTQ5MTEsIm5iZiI6MTcwOTI5NDYxMSwicGF0aCI6Ii8xMTUwNjY2NC8zMDkyNTE5NzMtOTJiNjFlMTEtZTk5ZS00YjllLTliYmEtNTg3ZTg4NGJlMjM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzAxVDEyMDMzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZmMDk4OGFiNTM2MTM4MjA2YmI1MDJlNTgzY2E4MTU3ZGM2OTI0OWI0NTBmNWVjMTc5YTE3ZTgxODNhMzE2MDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Do9y60biLirnxcKZMqm9Fg-DqaOAo4PXVggKohmQ8b8)
> 
> Thus, there seem to be some missing files from extracting the original KITTI data. (Or some label files don't get written.)

Thanks for your work and your hint that this is indeed the reason: when I initially uncompressed the semantickitt dataset, there were missing files, which means I have been training with the missing dataset!🤣
